### PR TITLE
Fix stable_train_samples

### PR DESCRIPTION
**Pull request by Rohan138**
Fix how stable_train_samples is calculated. This is a ROCm/transformers specific change to add warmup before collecting perf numbers, but it is currently not working as expected. Specifically:
- https://github.com/ROCm/transformers/blob/main/src/transformers/trainer.py#L2347 skips the first 10 steps of training
- https://github.com/ROCm/transformers/blob/main/src/transformers/trainer.py#L2503 is *supposed* to skip the first 10 steps' samples, but [args.warmup_steps](https://github.com/ROCm/transformers/blob/main/src/transformers/training_args.py#L311) is actually intended for learning rate warmup and defaults to 0.

E.g. if batch_size is 10, total_steps is 150, first 10 steps take 2 seconds, next 140 steps take 1 second, then:
- train_samples_per_second = (10 * 150) / (10 * 2 + 140 * 1) = **9.375**
- stable_train_samples_per_second (expected) = (10 * 150 - 10 * 10) / (140 * 1) = **10.000**
- stable_train_samples_per_second (current) = (10 * 150) / (140 * 1) = **10.714**

Instead, I added a *stable_train_warmup_steps* argument (default=10) to perform as intended.
With this change, pyt_huggingface_gpt2 perf changes from 559.092 to 529.131 stable_train_samples_per_second

NOTE: This will affect HF perf for QA, execdb, etc.

### CODE REVIEW

**Review by ppalaniappan-amd — APPROVED** [review]
LGTM
